src/System.ObjectModel/src/System/Collections/ObjectModel/ObservableCollection.cs

```diff
@@ -99,6 +100,384 @@ public void Move(int oldIndex, int newIndex)
             MoveItem(oldIndex, newIndex);
         }
 
+
+        /// <summary>
+        /// Adds the elements of the specified collection to the end of the <see cref="ObservableCollection{T}"/>.
+        /// </summary>
+        /// <param name="collection">
+        /// The collection whose elements should be added to the end of the <see cref="ObservableCollection{T}"/>.
+        /// The collection itself cannot be null, but it can contain elements that are null, if type T is a reference type.
+        /// </param>
+        /// <exception cref="ArgumentNullException"><paramref name="collection"/> is null.</exception>
+        public void AddRange(IEnumerable<T> collection)
+        {
+            InsertRange(Count, collection);
+        }
+
+        /// <summary>
+        /// Inserts the elements of a collection into the <see cref="ObservableCollection{T}"/> at the specified index.
+        /// </summary>
+        /// <param name="index">The zero-based index at which the new elements should be inserted.</param>
+        /// <param name="collection">The collection whose elements should be inserted into the List<T>.
+        /// The collection itself cannot be null, but it can contain elements that are null, if type T is a reference type.</param>                
+        /// <exception cref="ArgumentNullException"><paramref name="collection"/> is null.</exception>
+        /// <exception cref="ArgumentOutOfRangeException"><paramref name="index"/> is not in the collection range.</exception>
+        public void InsertRange(int index, IEnumerable<T> collection)
+        {
+            if (collection == null)
+                throw new ArgumentNullException(nameof(collection));
+
+            if (index > Count)
+                throw new ArgumentOutOfRangeException(nameof(index));
+
+            if (collection is ICollection<T> list)
+            {
+                if (list.Count == 0)
+                {
+                    return;
+                }
+                if (!(list is IList))
+                {
+                    list = new List<T>(list);
+                }
+            }
+            else if (!ContainsAny(collection))
+            {
+                return;
+            }
+            else
+            {
+                list = new List<T>(collection);
```

Comment:
Unnesessary  copying  if `collection` implements `IList` but not `ICollection<T>`.

Reply:
Will fix.

Reply:
Fixed.